mypy/semanal.py

```diff
@@ -3251,12 +3251,15 @@ def name_already_defined(self, name: str, ctx: Context) -> None:
         self.fail("Name '{}' already defined".format(name), ctx)
 
     def fail(self, msg: str, ctx: Context, serious: bool = False, *,
-             blocker: bool = False) -> None:
+             blocker: bool = False, implicit_any: bool = False) -> None:
```

Comment:
It seems like adding an extra parameter to fail function might not be the best way to do it.
I would try to keep this function generic for any kind of failures. 

Perhaps, you can check if the error is in the caller.